langchain/agents/mrkl/output_parser.py

```diff
@@ -22,6 +22,9 @@ def parse(self, text: str) -> Union[AgentAction, AgentFinish]:
             r"Action\s*\d*\s*:[\s]*(.*?)[\s]*Action\s*\d*\s*Input\s*\d*\s*:[\s]*(.*)"
         )
         match = re.search(regex, text, re.DOTALL)
+        if not match:  # try again with another format as LLM output is not stable
```

Comment:
this seems a bit separate? could we handle/discuss in a separate pr?

Reply:
@hwchase17 Sure. I just removed it.
cc @skcoirz Please create another PR for the parser changes and explain why we need it.